src/pages/CompanyMaterialPage/index.tsx

```diff
@@ -1,24 +1,50 @@
 import React, { useEffect, useState } from 'react'
 import MaterialTable from './MaterialTable'
 import FileChooser from './FileChooser'
-import { getFilesByCreator } from '../../services/companyMaterialService'
-import { CompanyFile } from '../../types/types'
+import { getFilesByCreator, getFilesById } from '../../services/companyMaterialService'
+import contracsService from '../../services/contractsService'
+import { useSelector } from 'react-redux'
+import { CompanyFile, roles } from '../../types/types'
+import { IRootState } from '../../utils/store'
 
 const CompanyMaterialsPage: React.FC = () => {
   const [files, setFiles] = useState<CompanyFile[]>([])
+  const { data } = useSelector((state: IRootState) => state.user)
+  const role = data.role
 
-  useEffect(() => {
-    const fetchFiles = async () => {
-      const filesFromServer = await getFilesByCreator()
-      setFiles(filesFromServer)
-    }
+  {
+    role === roles.Worker &&
+      useEffect(() => {
+        let filesFromServer: CompanyFile[] = []
 
-    fetchFiles()
-  }, [])
+        const fetchFiles = async () => {
+          const agreementsFromServer =
+            await contracsService.fetchEmploymentContractsAsWorkerOrBusiness()
+
+          for (let i in agreementsFromServer) {
```

Comment:
```suggestion
          for (const i in agreementsFromServer) {
```
Tässä voi myös käyttää const